docs/guides/connecting_to_your_data/database/postgres.md

```diff
@@ -0,0 +1,146 @@
+---
+title: How to connect to your data in a postgresql database
+---
+import AddingCredentials from '../reference/database_credentials.md'
+import NextSteps from '../reference/link-to-validator.md'
+import ConnectionStringDetails from '../reference/configuration_explanation.md'
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+This guide will help you connect Great Expectations to data in a Postgresql database. We will be building a `Datasource` configuration, and using it to verify connectivity. By the end of this how-to guide, you should have a `Batch` representing a slice of data in your database, and a working code snippet that can be used as a starting point for more complex configurations.
+
+:::note Prerequisites: This how-to-guide assumes you have already:
+- Completed the [Getting Started Tutorial.](../../tutorials/quick-start.md)
+- Have a working installation of Great Expectations.
+- Have data on Postgresql database that you would like to connect to.
+:::
+
+## Steps
+
+### 1. Determine how to add credentials to configuration
+
+Great Expectations provides multiple methods of providing credentials for accessing databases. For our current example will use a `connection_string`, but other options include providing an Environment Variable, and loading from a Cloud Secret Store.  
+
+For more information, please refer to [Additional Notes](#additional-notes).
+
+```
+postgresql+psycopg2://postgres:@localhost/test_ci
+```
+
+<details><summary><b>What's in the config?</b></summary>
+<p>
+<ConnectionStringDetails />
+</p>
+</details>
+
+### 2. Install required packages
+
+Next make sure you have the necessary packages for Great Expectations to connect to your postgres database.
+
+```console
+pip install sqlalchemy psycopg2
+```
+
+### 3. Load the DataContext into memory
+
+<details><summary><b>More Details</b></summary>
+<p>
+Open up a Jupyter Notebook in the same directory as the `great_expectations/` folder. Import any necessary packages or modules.
+
+```python file=../../../../integration/code/connecting_to_your_data/database/postgres.py#L1-L4
+```
+
+</p>
+</details>
+
+Load your DataContext into memory using the `get_context()` method.
+
+```python file=../../../../integration/code/connecting_to_your_data/database/postgres.py#L13
+```
+
+### 4. Write your configuration as a YAML
+
+Here is an example configuration:
+
+```python file=../../../../integration/code/connecting_to_your_data/database/postgres.py#L19-L33
```

Comment:
This snippet also looks like the line numbers have shifted. Please correct.

Reply:
Thank you.. I dont know what happened with these..